Remotion/Web/Core/ExecutionEngine/WxeUrlSettings.cs

```diff
@@ -0,0 +1,83 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Remotion.ServiceLocation;
+
+namespace Remotion.Web.ExecutionEngine
+{
+  /// <summary>
+  /// Contains relevant settings for the URL functionality of the web execution engine.
+  /// </summary>
+  [ImplementationFor(typeof(WxeUrlSettings), Lifetime = LifetimeKind.Singleton)]
+  public class WxeUrlSettings
+  {
+    private const string c_defaultUrlMappingFile = "UrlMapping.xml";
+
+    private const int c_defaultMaximumUrlLength = 1024;
+
+    private const string c_defaultDefaultWxeHandler = "";
```

Comment:
As for the only usecase in production code, we use IsNullOrEmpty for some fallback & exception, OR the actual value. Seems fine to me? I'm not sure if it would make sense to have an empty value be something different than null